python/GafferSceneUI/AttributeTweaksUI.py

```diff
@@ -104,56 +104,6 @@
 	}
 )
 
-##########################################################################
-# Internal utilities
-##########################################################################
-
-def _attributeTweaksNode( plugValueWidget ) :
-
-	# The plug may not belong to an AttributeTweaks node
-	# directly. Instead it may have been promoted
-	# elsewhere and be driving a target plug on an
-	# AttributeTweaks node.
-
-	def walkOutputs( plug ) :
-
-		if isinstance( plug.node(), GafferScene.AttributeTweaks ) :
-			return plug.node()
-
-		for output in plug.outputs() :
-			node = walkOutputs( output )
-			if node is not None :
-				return node
-
-	return walkOutputs( plugValueWidget.getPlug() )
```

Comment:
Are we losing something valuable by not following the plug to its destination for the attribute popup? The `tweaks` plug has to be promoted via Python, so I don't know how common it is to promote it, but it's possible to do.

Reply:
The stuff I removed was only used by the private `_TweaksFooter` class, which doesn't go along for the ride when the plug is promoted. I was in two minds about what to do here, but went with the principle that there's no point maintaining code that isn't currently needed. I could be persuaded the other way if you feel strongly, or we have evidence of people doing a cheeky promotion of the private widget.

Reply:
Summary off out-of-band chat : OK to omit all this stuff, because the UI is private and is highly unlikely to have been deliberately promoted to another plug.